core/services/keeper/orm.go

```diff
@@ -134,19 +134,27 @@ func (korm ORM) SetLastRunHeightForUpkeepOnJob(db *gorm.DB, jobID int32, upkeepI
 		).Error
 }
 
-func (korm ORM) CreateEthTransactionForUpkeep(sqlDB *sql.DB, upkeep UpkeepRegistration, payload []byte, maxUnconfirmedTXs uint64) (models.EthTx, error) {
+func (korm ORM) CreateEthTransactionForUpkeep(tx *gorm.DB, upkeep UpkeepRegistration, payload []byte, maxUnconfirmedTXs uint64) (models.EthTx, error) {
 	var etx models.EthTx
 	ctx, cancel := postgres.DefaultQueryCtx()
 	defer cancel()
 
+	sqlTx, ok := tx.Statement.ConnPool.(*sql.Tx)
+	if !ok {
+		return etx, errors.New("unable to get tx from conn pool")
+	}
+	sqlDB, err := tx.DB()
+	if err != nil {
+		return etx, err
+	}
```

Comment:
Would we ever actually expect this to fail?

Might be nice to have a `postgres.MustSQLDBFromTx` that panics on fail

Reply:
it can fail if you don't open a tx before calling it. I'm fine panicking, I guess it'd raise the issue earlier in the dev cycle if you were not passing in a tx